llama_index/schema.py

```diff
@@ -9,6 +9,18 @@
 from llama_index.utils import get_new_id
 
 
+def _validate_is_flat_dict(metadata_dict: dict) -> None:
+    """
+    Validate that metadata dict is flat, 
+    and only uses str keys, and (str, int, float) values.
+    """
+    for key, val in metadata_dict.items():
+        if not isinstance(key, str):
+            raise ValueError('Metadata key must be str!')
+        if not isinstance(val, (str, int, float)):
+            raise ValueError('Value must be one of (str, int, float)')
```

Comment:
nit: "Value" -> "Metadata dict values"

Reply:
I see in the pinecone notebook, you set the theme to be a list in extra_info. Wouldn't this trigger this error? Or this requirement is only for documents?